### ISSUE TITLE
Missing open brace gets indentation error

### ISSUE THREAD

**Issue by davecparker** [davecparker/code12#204]
The following code and similar code results in an indentation error:
"A block's ending } should have the same indentation as its beginning {"
referencing the open { of the function.
```
void foo()
{
    if (x > 0)
        x = 0;
    }
    y = 0;
}
```
But the user's mistake is really that they left out a { for the if, or had an unneeded } at the end.
(The user fixing it as an indentation error as reported then results in a bigger error).

Maybe you can ponder how to give a better error and heuristically determine when to do so, depending maybe on the number of controlled stmts, looking ahead to the line after the }, etc.

**Comment by davecparker** [comment]
Oops, code in the original report what not what I intended. Edited above.
The case with two controlled stmts is not as bad, but also interesting to ponder if you can improve it.

**Comment by bcauchon** [comment]
How's this look?
![image](https://user-images.githubusercontent.com/38901147/45002916-27f0a080-af91-11e8-9eb3-83582e4e2cde.png)
![image](https://user-images.githubusercontent.com/38901147/45002932-6be3a580-af91-11e8-82f5-f83547c17a72.png)

**Comment by bcauchon** [comment]
What do you think of this for the two controlled stmts? I'm not checking for the closing }, just that the two statements have the same indentation.
![image](https://user-images.githubusercontent.com/38901147/45003701-09db6e00-af9a-11e8-8b5f-33880e3816ab.png)
I can also do more work to look for a closing } with the right indentation to suggest a missing opening { if you think that would be better.

**Comment by davecparker** [comment]
I like the error for the two controlled stmts ("not controlled by the if") a lot, that aligns well with rookie mistakes I think, nice. 

The first case ("expected less indentation") is an improvement, but I wonder if it can be made better by looking at the fact that the indentation of the } matches the if and that there is only one controlled stmt, meaning that maybe the } is extra/stray, not that a { is missing. But it is still a probability guessing what the mistake really was... In the case where there is another stmt after the } at the same level (here y = 0;) raises the probability that the } is not closing the parent block... Especially if the parent block is a function, since stmts don't belong outside functions... Fodder to think about, maybe not worth a huge amount of work if it's too hard.

**Comment by bcauchon** [comment]
For the multi-statement case I have it now so that if there is a closing bracket and it looks like a block from the control statement to the bracket you get this error message:
![image](https://user-images.githubusercontent.com/38901147/45004456-ac4b1f80-afa1-11e8-850a-ddae06175573.png)
I would like to highlight the controlling stmt and the ending { bracket like in the single statement case, but then parseProgram keeps running and the "unexpected change in indentation" error is set on an earlier line above the ending { bracket and that error is displayed instead.

Otherwise if there is no ending } bracket or the in-between code's indentation doesn't make it look like a block you get this error:
![image](https://user-images.githubusercontent.com/38901147/45004577-b883ac80-afa2-11e8-8c11-81a4d3b8ed0d.png)

**Comment by bcauchon** [comment]
I just saw your last reply after I posted the one above. Maybe it's better not to highlight the } bracket.

How about this for the single statement case?
![image](https://user-images.githubusercontent.com/38901147/45004784-46ac6280-afa4-11e8-8ead-990bcd59148f.png)

**Comment by davecparker** [comment]
That might work. I assume the possibilities are starting to interact, so it's probably time to give the change a shot and start playing with a bunch of different possibilities and see what happens.

**Comment by bcauchon** [comment]
Fixed in #205
